publisher/lib/ontology_collector.py

```diff
@@ -1,106 +1,78 @@
 import argparse
-import os
+import os.path
+import re
+import sys
 
 from rdflib import Graph, OWL, RDF
 
+visited_ontologies = list()
+local_ontology_map_regex = re.compile('name="(.+)"\s+uri="(.+)"')
+local_ontology_map = dict()
+global output_graph
+
+
+def get_local_ontology_map(ontology_catalog_path: str):
+    try:
+        ontology_catalog_file = open(ontology_catalog_path, 'r')
+        ontology_catalog = ontology_catalog_file.read()
+        ontologies = local_ontology_map_regex.findall(string=ontology_catalog)
+        for ontology in ontologies:
+            local_ontology_map[ontology[0]]=ontology[1][2:]
+            v=local_ontology_map
+    except Exception as exception:
+        print(str(exception))
+        sys.exit(-1)
+    
+    
+def collect_ontologies(
+        root: str,
+        input_ontology_path: str,
+        output_graph_path=str(),
+        save_output=False):
+    global output_graph
```

Comment:
Using global variables isn't considered a good practice.

Reply:
I would like to keep the code here as simple as possible